### DISCUSSION TITLE
Error when building webvault

### DISCUSSION THREAD

**Discussion by jfreak53** [dani-garcia/vaultwarden#4249] [Q&A] [answered]
I'm trying to build the webvault, all else is working fine. I followed instructions [here](https://github.com/dani-garcia/vaultwarden/wiki/Building-binary#install-the-web-vault), but no go.

```
Checking patch libs/components/src/menu/menu.component.html...
error: while searching for:
<ng-template>
  <div
    (click)="closed.emit()"
    class="tw-flex tw-shrink-0 tw-flex-col tw-rounded tw-border tw-border-solid tw-border-secondary-500 tw-bg-background tw-bg-clip-padding tw-py-2"
    [attr.role]="ariaRole"
    [attr.aria-label]="ariaLabel"
    cdkTrapFocus

error: patch failed: libs/components/src/menu/menu.component.html:1
Applied patch apps/web/src/app/admin-console/organizations/create/organization-information.component.html cleanly.
Applied patch apps/web/src/app/admin-console/organizations/settings/account.component.html cleanly.
Applied patch apps/web/src/app/admin-console/organizations/settings/account.component.ts cleanly.
Applied patch apps/web/src/app/app.component.ts cleanly.
Applied patch apps/web/src/app/auth/settings/two-factor-authenticator.component.ts cleanly.
Applied patch apps/web/src/app/billing/organizations/organization-billing-history-view.component.ts cleanly.
Applied patch apps/web/src/app/billing/organizations/organization-plans.component.html cleanly.
Applied patch apps/web/src/app/billing/organizations/organization-plans.component.ts cleanly.
Applied patch apps/web/src/app/components/environment-selector/environment-selector.component.ts cleanly.
Applied patch apps/web/src/app/core/init.service.ts cleanly.
Applied patch apps/web/src/app/core/router.service.ts cleanly.
Applied patch apps/web/src/app/core/web-platform-utils.service.ts cleanly.
Applied patch apps/web/src/app/layouts/footer.component.html cleanly.
Applied patch apps/web/src/app/layouts/frontend-layout.component.html cleanly.
Applied patch apps/web/src/app/layouts/header/web-header.component.html cleanly.
Applied patch apps/web/src/app/layouts/navbar.component.html cleanly.
Applied patch apps/web/src/app/settings/settings.component.ts cleanly.
Applied patch apps/web/src/app/tools/generator.component.ts cleanly.
Applied patch apps/web/src/app/tools/send/access.component.html cleanly.
Applied patch apps/web/src/app/vault/individual-vault/vault.component.ts cleanly.
Applied patch apps/web/src/app/vault/org-vault/vault-header/vault-header.component.ts cleanly.
Applied patch apps/web/src/app/vault/org-vault/vault.component.ts cleanly.
Applied patch apps/web/src/index.html cleanly.
Applied patch apps/web/src/manifest.json cleanly.
Applied patch apps/web/src/scss/styles.scss cleanly.
Applied patch apps/web/webpack.config.js cleanly.
Applied patch libs/angular/src/auth/components/register.component.ts cleanly.
Applied patch libs/angular/src/auth/components/two-factor-options.component.ts cleanly.
Applying patch libs/components/src/menu/menu.component.html with 1 reject...
Rejected hunk #1.
FAILED: line 33, exit code 1. Press any key to exit...
Makefile:44: recipe for target 'patch-web-vault' failed
make: *** [patch-web-vault] Error 1
```

**Comment by BlackDex** [comment]
You are probably using the `main` branch of the Bitwarden client repo.
In that case they merged a PR from me which has that specific patch already implemented.

If you use anything else then the versions we have patches for, you need to create a new patch-file your self and make sure it has a higher version number, or provide the specific name of the patch.

**Reply by jfreak53** [reply]
I mean I haven't done anything to it other than what the directions said. For the client I literally just cloned vaultwarden, ran a build, and now need the web client. Again, no changes from me.

So do I need to do anything other than clone it into where I have the VW binary? I'm lost as to what needs done, sorry.

**Reply by BlackDex** [reply, marked as answer]
Then why build it your self? And not use docker?
Or the tar.gz files from the bw_web_build release page?

**Reply by jfreak53** [reply]
Old VM, ubu 16, can't use docker. I will download the release then. I was simply following the guides on your repo in the order I found them, sorry.

**Reply by BlackDex** [reply]
The only thing i can think of, is that you didn't checked out the correct tag/hash to match with the latest patch/version we support.

Maybe if you share the exact commands used we can see what went wrong, and how to improve the wiki.

**Reply by jfreak53** [reply]
Downloading release worked fine :) thanks

**Reply by jfreak53** [reply]
One last question, in the old version I had `SERVER_ADMIN_EMAIL` for a super admin that would be able to manage users and such. I logged in as this user just now, and no longer has access to the `rs` organization to manage users. What do I do now for that?

**Reply by BlackDex** [reply]
There never was something like that.
I think you are mixing Vaultwarden with Bitwarden.

**Reply by jfreak53** [reply]
There was though, I had bitwarden rs and upgraded to vaultwarden. It was
the only way to manage users.

**Reply by jfreak53** [reply]
https://github.com/dani-garcia/vaultwarden/issues/249

Use to exist, how do we do this now?

**Reply by BlackDex** [reply]
To be fair, that has been removed 6 years ago already!
It was that old that i didn't remembered that at all.

Read the wiki regarding the admin interface https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token

**Reply by jfreak53** [reply]
Hahaha sorry, I couldn't find anything on admin in my searching and I don't upgrade all that often. Thank you!
